rust/crates/langsmith-pyo3/src/py_run.rs

```diff
@@ -196,6 +213,45 @@ impl FromPyObject<'_> for RunCommon {
     }
 }
 
+fn extract_attachments(value: &Bound<'_, PyAny>) -> PyResult<Option<Vec<Attachment>>> {
+    if value.is_none() {
+        return Ok(None);
+    }
+
+    let mapping = value.downcast::<PyMapping>()?;
+
+    let size = mapping.len()?;
+    if size == 0 {
+        return Ok(None);
+    }
+
+    let mut attachments = Vec::with_capacity(size);
+
+    for result in mapping.items()?.iter()? {
+        let key_value_pair = result?;
+
+        let key_item = key_value_pair.get_item(0)?;
+        let key = key_item.extract::<&str>()?;
+
+        // Each value in the attachments dict is a (mime_type, bytes) tuple.
+        let value = key_value_pair.get_item(1)?;
+        let value_tuple = value.downcast_exact::<PyTuple>()?;
+        let mime_type_value = value_tuple.get_item(0)?;
+        let bytes_value = value_tuple.get_item(1)?;
+
+        attachments.push(Attachment {
+            // TODO: It's unclear whether the key in the attachments dict is
+            //       the `filename`` or the `ref_name`, and where the other one is coming from.
+            ref_name: key.to_string(),
+            filename: key.to_string(),
```

Comment:
it's the ref_name

Reply:
The attachments dict is `name -> (mime_type, data)`, so is using the key for both `ref_name` and `filename` the right thing to do?